### ISSUE TITLE
Bar suggesting desktop notifications disappears & causes side panel glitches

### ISSUE THREAD

**Issue by aperezdc** [vector-im/riot-web#3866]
### Description

When opening Riot, and the permission to use desktop notifications was not yet given by the user, the bar that appears at the top suggesting to enable them sometimes disappears (seems to happen always in rooms, but not when opening the directory). Once that top bar disappears, a visual glitch in the sidebars shifts their bottom side upwards.

This can be better understood with a couple of screenshots. The bar at the top which suggests enabling notifications is this one:

![notifs-enable-topbar](https://cloud.githubusercontent.com/assets/723451/25861528/b3330d60-34ed-11e7-8b43-39a2d6126ba9.png)

When it disappears, the sidebars may be left looking like in the following capture:

![riot-notifs-topbar-missing-glitch](https://cloud.githubusercontent.com/assets/723451/25861676/1a51dabc-34ee-11e7-84fb-4781f1dbe843.png)

Notice the top bar with the suggestion has disappeared, and at the bottom of both side panels there is now some empty space (where the red arrows are). This empty space seems to have a vertical size similar to the bar that disappeared.

### Steps to reproduce

The easiest way for me to reproduce this was using [Epiphany](https://wiki.gnome.org/Apps/Web) 3.24.x from the command line, and using a separate profile directory.

The steps are:

- From the terminal, run Epiphany creating a new profile directory with `epiphany --profile=$HOME/riot-test`.
- Log-in to Matrix, enable desktop notifications when suggested.
- Navigate to a room. Copy the room URL to the clipboard (`Ctrl+L` followed by `Ctrl+C` will do).
- Close Epiphany and remove the saved permissions file: `rm $HOME/riot-test/permissions.ini`
- Run Epiphany again, passing the URL to the room from the clipboard: `epiphany --profile=$HOME/riot-test <URL>`.

Most of the time, top bar won't show (despite the permission to use desktop notifications being missing) and the side panels will show up as in the screenshot above. Using the above steps with Riot from the `develop` branch seems to always trigger this glitch.

### Version information

- **Platform**: Web.

For the web app:

- **Browser**: Epiphany (based on WebKitGTK+), probably others. Seems to happen with others as well [according to Yaniel](https://matrix.to/#/!DgvjtOljKujDBrxyHk:matrix.org/$149434634564704NPKwu:kolm.io)
- **OS**: Arch Linux
- **URL**: riot.im/develop

**Comment by lampholder** [comment]
Thanks for the detailed report :)

P1ing this (perhaps a little enthusiastically) because the nag to turn on desktop notifications is likely something that very new users will see, so this is a particularly prominent broken window.

**Comment by aperezdc** [comment]
I did a bit of investigation on this. It's happening both in `/develop` and stable (`/app`). From the inspector I could check that the top bar element is still there (good) and in the `mx_MatrixChat` element there is the added class `mx_MatricChat_toolbarShowing`. Wiggling the height of the window I've been able to momentarily make the bar appear:

![inspector](https://cloud.githubusercontent.com/assets/723451/25946782/4300d18a-3655-11e7-8611-ffb883723260.png)

This used to work fine with a couple stable releases ago. Let me know if you would like me check older stable releases and/or bisect the issue.

**Comment by t3chguy** [comment]
Closing assuming this hasn't been seen in years. Re-open if anyone hits this again.
